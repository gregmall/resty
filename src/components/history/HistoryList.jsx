import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';


const HistoryList = ({ history }) => {
  console.log(history)
  return history.map(result => (
    <>
    <h2>History</h2>
      <HistoryItem key ={ result.url + result.method }  {...result}/>
    </>
  ));

}
History.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired
  }))
}

export default HistoryList;
