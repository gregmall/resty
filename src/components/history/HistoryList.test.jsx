import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList'

describe('HistoryItem component', () => {
  afterAll(() => cleanup());
  
  it('displays a history item', () => {
    const { asFragment } = render(<HistoryList
      history={[
        {
          url: "www.website.com",
          method: "GET"
        },
        { 
          url: "www.another.com",
          method: "PUT"

        },
        {
          url: "www.there.com",
          method: "DELETE"
        }
      ]}
    
      />);
      expect(asFragment()).toMatchSnapshot();
  })
})
