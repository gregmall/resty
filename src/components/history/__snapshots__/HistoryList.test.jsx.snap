// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component displays a history item 1`] = `
<DocumentFragment>
  <section
    class="HistoryItem"
  >
    <p>
      Method: GET
    </p>
    <p>
        Url: www.website.com
    </p>
  </section>
  <section
    class="HistoryItem"
  >
    <p>
      Method: PUT
    </p>
    <p>
        Url: www.another.com
    </p>
  </section>
  <section
    class="HistoryItem"
  >
    <p>
      Method: DELETE
    </p>
    <p>
        Url: www.there.com
    </p>
  </section>
</DocumentFragment>
`;
